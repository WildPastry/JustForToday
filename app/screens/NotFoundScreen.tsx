import { ColorSchemeName, StyleSheet, TouchableOpacity } from 'react-native';
import { Text, View } from '../components/styles/Themed';
import Colours from '../constants/Colours';
import Control from '../constants/Control';
import { IDeviceSize } from '../types/generic.types';
import { MaterialIcons } from '@expo/vector-icons';
import { RootStackScreenProps } from '../types/navigation.types';
import getDeviceSize from '../constants/Layout';
import useColorScheme from '../hooks/useColorScheme';

export default function NotFoundScreen({
  navigation
}: RootStackScreenProps<'NotFound'>) {
  // Screen settings
  const colorScheme: NonNullable<ColorSchemeName> = useColorScheme();
  const deviceSize: IDeviceSize[keyof IDeviceSize] = getDeviceSize();

  return (
    <View style={[styles.container, Control[deviceSize].container]}>
      <View style={styles.notFoundContainer}>
        <MaterialIcons
          style={styles.notFoundIcon}
          name='error'
          size={Control[deviceSize].icon}
          color={Colours[colorScheme].text}
        />
        <Text style={Control[deviceSize].text}>This screen doesn't exist.</Text>
      </View>
      <TouchableOpacity
        onPress={() => navigation.replace('Root')}
        style={styles.link}>
        <Text
          style={Control[deviceSize].text}
          lightColor={Colours.light.link}
          darkColor={Colours.dark.link}>
          Go to home screen
        </Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center'
  },
  notFoundContainer: {
    flexDirection: 'row'
  },
  notFoundIcon: {
    paddingRight: 10
  },
  link: {
    marginTop: 30
  }
});
