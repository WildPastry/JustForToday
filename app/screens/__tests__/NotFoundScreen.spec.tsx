import NotFoundScreen from '../NotFoundScreen';
import renderer from 'react-test-renderer';

describe('<NotFoundScreen />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NotFoundScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
