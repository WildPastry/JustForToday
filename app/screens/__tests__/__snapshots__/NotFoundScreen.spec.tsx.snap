// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundScreen /> renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
      ]
    }
  >
    This screen doesn't exist.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "marginTop": 15,
        "opacity": 1,
        "paddingVertical": 15,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#000",
          },
          {
            "color": "#2e78b7",
            "fontSize": 14,
          },
        ]
      }
    >
      Go to home screen!
    </Text>
  </View>
</View>
`;
